python/cudf/cudf/core/frame.py

```diff
@@ -1821,6 +1829,249 @@ def sample(
+        cudf_non_category_frame = (
+            libcudf.table.Table()
+            if data.num_columns == 0
+            else libcudf.interop.from_arrow(data, data.column_names)
+        )
+
+        if (
+            cudf_non_category_frame._num_columns > 0
+            and cudf_category_frame._num_columns > 0
+        ):
+            result = cudf_non_category_frame
+            for name in cudf_category_frame._data.names:
+                result._data[name] = cudf_category_frame._data[name]
+        elif cudf_non_category_frame._num_columns > 0:
+            result = cudf_non_category_frame
+        else:
+            result = cudf_category_frame
+
+        if pandas_dtypes:
+            for name in result._data.names:
+                if pandas_dtypes[name] == "categorical":
+                    dtype = "category"
+                else:
+                    dtype = np_dtypes[name]
+
+                result._data[name] = result._data[name].astype(dtype)
+
+        result = libcudf.table.Table(
+            result._data.select_by_label(column_names)
+        )
+
+        return cls._from_table(result)
+
+    @annotate("TO_ARROW", color="orange", domain="cudf_python")
+    def to_arrow(self):
+        """
+        Convert to arrow Table
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame(
+        ...     {"a":[1, 2, 3], "b":[4, 5, 6]}, index=[1, 2, 3])
+        >>> df.to_arrow()
+        pyarrow.Table
+        a: int64
+        b: int64
+        index: int64
+        """
+
+        data = self.copy(deep=False)
+
+        codes = {}
+        # saving the name as they might get changed from int to str
+        codes_keys = []
+        categories = {}
+        # saving the name as they might get changed from int to str
+        names = self._data.names
+        null_arrays_names = []
+
+        for name in self._data.names:
+            col = self._data[name]
+            if isinstance(col, cudf.core.column.CategoricalColumn,):
+                # arrow doesn't support unsigned codes
+                signed_type = (
+                    min_signed_type(col.codes.max())
+                    if col.codes.size > 0
+                    else np.int8
+                )
+                codes[str(name)] = col.codes.astype(signed_type)
+                categories[str(name)] = col.categories
+                codes_keys.append(name)
+            elif isinstance(
+                col, cudf.core.column.StringColumn
+            ) and col.null_count == len(col):
+                null_arrays_names.append(name)
+
+        data = cudf.DataFrame(data._data)
+        data = data.drop(codes_keys + null_arrays_names, inplace=True)
+
+        out_table = pa.table([])
+        if data.shape[1] > 0:
+            out_table = libcudf.interop.to_arrow(
+                data, data._data.names, keep_index=False
+            )
+
+        if len(codes) > 0:
+            codes_df = cudf.DataFrame(codes)
```

Comment:
Similarly would be good to avoid using `cudf.DataFrame` here as well.

Reply:
missed this to create as Table rather than DataFrame.